cronback-services/src/metadata/handler.rs

```diff
@@ -135,11 +140,76 @@ impl MetadataSvc for MetadataSvcHandler {
         }))
     }
 
+    async fn get_project_notification_settings(
+        &self,
+        request: Request<GetProjectNotificationSettingsRequest>,
+    ) -> Result<Response<GetProjectNotificationSettingsResponse>, Status> {
+        let req = request.into_inner();
+        let project_id: ProjectId = req.id.unwrap().into();
+        let project_id = project_id
+            .validated()
+            .map_err(|e| Status::invalid_argument(e.to_string()))?;
+        let status = self
+            .project_store
+            .get_notification_settings(&project_id)
+            .await
+            .map_err(ProjectStoreHandlerError::Store)?;
+        match status {
+            | Some(st) => {
+                Ok(Response::new(GetProjectNotificationSettingsResponse {
+                    settings: Some(st.into()),
+                }))
+            }
+            | None => {
+                Err(ProjectStoreHandlerError::NotFound(format!(
+                    "{}",
+                    project_id
+                )))?
+            }
+        }
+    }
+
+    async fn set_project_notification_settings(
+        &self,
+        request: Request<SetProjectNotificationSettingsRequest>,
+    ) -> Result<Response<SetProjectNotificationSettingsResponse>, Status> {
+        let req = request.into_inner();
+        let project_id: ProjectId = req.id.unwrap().into();
+        let project_id = project_id
+            .validated()
+            .map_err(|e| Status::invalid_argument(e.to_string()))?;
+
+        let old_settings = self
+            .project_store
+            .get_notification_settings(&project_id)
+            .await
+            .map_err(ProjectStoreHandlerError::Store)?;
+
+        let Some(old_settings) = old_settings else {
+            return Err(ProjectStoreHandlerError::NotFound(
+                project_id.to_string(),
+            )
+            .into());
+        };
+
+        self.project_store
+            .set_notification_settings(
+                &project_id,
+                req.settings.unwrap().into(),
+            )
+            .await
+            .map_err(ProjectStoreHandlerError::Store)?;
```

Comment:
I think we should explore a better way for error handling, the number of .map_err is pretty high. I'll give it some thought.